code/game/machinery/kitchen/cooking_machines/oven.dm

```diff
@@ -50,24 +50,38 @@
 
 
 /obj/machinery/appliance/cooker/oven/AltClick(var/mob/user)
+	.=1
 	if(user.stat || user.restrained())	return
 	user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)//No spamming the door, it makes a sound
 	toggle_door()
 
+
 /obj/machinery/appliance/cooker/oven/verb/toggle_door()
-	set src in view()
+	set src in world //So that AI can operate it remotely
```

Comment:
It does not appear in the rightclick menu without doing this. This is what i had to do to make it appear there